pyexchange/leverj.py

```diff
@@ -375,39 +376,50 @@ def approve_token(self, token_address: str, amount: int) -> Transact:
     def deposit_ether(self, leverjobj: LeverjAPI, amount: Wad, gluon_block_number):
         assert(isinstance(leverjobj, LeverjAPI))
         assert(isinstance(amount, Wad))
-        assert(isinstance(gluon_block_number, None) or isinstance(gluon_block_number, int))
 
         custodian_account = self.address
         app_id = leverjobj.get_spot_exchange_id()
         if gluon_block_number is None:
-            gluon_block_number = leverjobj._http_authenticated("GET", "/api/v1", f"/plasma/{app_id}", None)['number'] +2
-            Transact(self, self.web3, self.abi, self.address, self._contract, "depositEther",[app_id], {'value': int(amount.value)}).transact()
-            return gluon_block_number
+            gluon_block_number = leverjobj._http_authenticated("GET", "/api/v1", f"/plasma/{app_id}", None)['number'] + 2
+            receipt = Transact(self, self.web3, self.abi, self.address, self._contract, "depositEther",[app_id], {'value': int(amount.value)}).transact()
```

Comment:
do we want just to return receipt or to check here if transaction completed ok / other details?

Reply:
we need to return receipt in it's full form to be checked by inventory service with gluon block to know where we are in the processing